### DISCUSSION TITLE
npx renovate fails with "no such file or directory, chmod '/__w/15/s/node_modules/graph-data-structure/dist/cli.module.js'"

### DISCUSSION THREAD

**Discussion by flcdrg** [renovatebot/renovate#20029] [Request Help] [answered]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

34.113.0

### If you're self-hosting Renovate, select which platform you are using.

Azure DevOps (dev.azure.com)

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

Be able to run renovate

### What you tried so far.

Ran `npx renovate`

Fails with:

```text
npm ERR! code ENOENT
npm ERR! syscall chmod
npm ERR! path /home/node/.npm/_npx/152/lib/node_modules/renovate/node_modules/graph-data-structure/dist/cli.module.js
npm ERR! errno -2
npm ERR! enoent ENOENT: no such file or directory, chmod '/home/node/.npm/_npx/152/lib/node_modules/renovate/node_modules/graph-data-structure/dist/cli.module.js'
npm ERR! enoent This is related to npm not being able to find a file.
npm ERR! enoent 
```

Also tried running `npm install renovate / npm run renovate` and got a similar error.

```text
npm ERR! code ENOENT
npm ERR! syscall chmod
npm ERR! path /__w/15/s/node_modules/graph-data-structure/dist/cli.module.js
npm ERR! errno -2
npm ERR! enoent ENOENT: no such file or directory, chmod '/__w/15/s/node_modules/graph-data-structure/dist/cli.module.js'
npm ERR! enoent This is related to npm not being able to find a file.
npm ERR! enoent
```

Runing an ls on the node_modules directory, and I can confirm that the graph-data-structure package is missing.

![image](https://user-images.githubusercontent.com/384747/214472906-99ffda72-b0e8-4e10-9a9f-bbbbafdf3027.png)

Also, the output of `npm version`

```json
{
  npm: '6.14.17',
  ares: '1.18.1',
  brotli: '1.0.9',
  cldr: '40.0',
  icu: '70.1',
  llhttp: '2.1.6',
  modules: '83',
  napi: '8',
  nghttp2: '1.42.0',
  node: '14.21.1',
  openssl: '1.1.1q',
  tz: '2022e',
  unicode: '14.0',
  uv: '1.42.0',
  v8: '8.4.371.23-node.87',
  zlib: '1.2.11'
}
```

**Comment by flcdrg** [comment, marked as answer]
I think this may be a problem with `npm`? Even though the node version complies with the minimum version as documented at https://docs.renovatebot.com/getting-started/running/#npm-package-cli

If I add the following step to my pipeline, it does download the graph-data-structure package correctly.

```yaml
  - task: UseNode@1
    inputs:
      version: 18.x
```

This gives me npm v8.19.3 (and node v18.13.0)

**Reply by rarkins** [reply]
The error you pasted is definitely an _npm_ error, and not a _renovate_ error message. It happens before renovate is even installed/loaded.
